eth/watchers/timewatcher.go

```diff
@@ -54,12 +54,44 @@ func NewTimeWatcher(roundsManagerAddr ethcommon.Address, watcher BlockWatcher, l
 		return nil, fmt.Errorf("error creating decoder: %v", err)
 	}
 
-	return &TimeWatcher{
+	tw := &TimeWatcher{
 		quit:    make(chan struct{}),
 		watcher: watcher,
 		lpEth:   lpEth,
 		dec:     dec,
-	}, nil
+	}
+
+	lr, err := tw.lpEth.LastInitializedRound()
+	if err != nil {
+		return nil, fmt.Errorf("error fetching initial lastInitializedRound value err=%v", err)
+	}
+	bh, err := tw.lpEth.BlockHashForRound(lr)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching initial lastInitializedBlockHash value err=%v", err)
+	}
+	num, err := tw.lpEth.CurrentRoundStartBlock()
+	if err != nil {
+		return nil, fmt.Errorf("error fetching current round start block %v=", err)
```

Comment:
I think this typo was here before, but look likes %v= should be err=%v.

Reply:
The related unit test needs to be fixed as well.

Reply:
Missed that, should be fixed now